File: app/src/main/java/com/ginger/democountryquizapp/utils/FileUtils.java
package com.ginger.democountryquizapp.utils;

import android.content.Context;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

public class FileUtils {

    public static String readJsonFromAssets(Context context, String fileName) {
        String json = null;
        try {
            InputStream is = context.getAssets().open(fileName);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, StandardCharsets.UTF_8);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }
}

File: app/src/main/java/com/ginger/democountryquizapp/utils/FileUtils.kt
package com.ginger.democountryquizapp.utils

import android.content.Context
import java.io.IOException
import java.nio.charset.StandardCharsets

object FileUtils {

fun readJsonFromAssets(context: Context, fileName: String): String? {
        return try {
val inputStream = context.assets.open(fileName)
val size = inputStream.available()
val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()
String(buffer, StandardCharsets.UTF_8)
        } catch (e: IOException) {
        e.printStackTrace()
            null
            }
            }
            }
